client/providers/github_actions.go

```diff
@@ -135,3 +140,25 @@ func (g *GithubOp) RequestTokens(ctx context.Context, cicHash string) (*memguard
 	// json.RawMessage leaves the " (quotes) on the string. We need to remove the quotes
 	return memguard.NewBufferFromBytes(jwt.Value[1 : len(jwt.Value)-1]), nil
 }
+
+func (g *GithubOp) RequestTokens(ctx context.Context, cic *clientinstance.Claims) ([]byte, error) {
+	// Define our commitment as the hash of the client instance claims
+	commitment, err := cic.Hash()
+	if err != nil {
+		return nil, fmt.Errorf("error calculating client instance claim commitment: %w", err)
+	}
+
+	// Use the commitment nonce to complete the OIDC flow and get an ID token from the provider
+	idTokenLB, err := g.requestTokens(ctx, string(commitment))
+	// idTokenLB is the ID Token in a memguard LockedBuffer, this is done
+	// because the ID Token contains the OPs RSA signature which is a secret
+	// in GQ signatures. For non-GQ signatures OPs RSA signature is considered
+	// a public value.
+	if err != nil {
+		return nil, fmt.Errorf("error requesting ID Token: %w", err)
+	}
+	defer idTokenLB.Destroy()
+	gqToken, err := CreateGQToken(ctx, idTokenLB.Bytes(), g)
+
+	return gqToken, err
```

Comment:
Preventing githubOP from generating a non-GQ signature is an intentional design choice. The concern is that GHA ID Tokens are dangerous enough that we want to protect people from leaking a non-GQ ID Token. If they really want to do this, then they should have to implement an OpenIdProvider to activate that foot gun.

I could be convinced otherwise if presented with a powerful usecase for non-GQ github ID Tokens. Right now it seems like non-GQ github ID Tokens are all risk and no additional benefit.

Previously we had  here to ensure that the client would error out if a user attempted to generate a non-GQ Github ID Token. Now we enforce it in the RequestTokens which is much nicer.

```golang
func (*GithubOp) VerifyNonGQSig(context.Context, []byte, string) error {
	return client.ErrNonGQUnsupported
}
```

https://github.com/openpubkey/openpubkey/blame/e67a03098be79ed2e27366bf8705f8e56a3c279c/client/providers/github_actions.go#L130-L132